build-locally.sh

```diff
@@ -0,0 +1,127 @@
+
+
+#--------------------------------------------------------------------------
+#
+# Set Colors
+#
+#--------------------------------------------------------------------------
+bold=$(tput bold)
+underline=$(tput sgr 0 1)
+reset=$(tput sgr0)
+
+red=$(tput setaf 1)
+green=$(tput setaf 76)
+white=$(tput setaf 7)
+tan=$(tput setaf 202)
+blue=$(tput setaf 25)
+
+#--------------------------------------------------------------------------
+#
+# Headers and Logging
+#
+#--------------------------------------------------------------------------
+underline() { printf "${underline}${bold}%s${reset}\n" "$@"
+}
+h1() { printf "\n${underline}${bold}${blue}%s${reset}\n" "$@"
+}
+h2() { printf "\n${underline}${bold}${white}%s${reset}\n" "$@"
+}
+debug() { printf "${white}%s${reset}\n" "$@"
+}
+info() { printf "${white}➜ %s${reset}\n" "$@"
+}
+success() { printf "${green}✔ %s${reset}\n" "$@"
+}
+error() { printf "${red}✖ %s${reset}\n" "$@"
+}
+warn() { printf "${tan}➜ %s${reset}\n" "$@"
+}
+bold() { printf "${bold}%s${reset}\n" "$@"
+}
+note() { printf "\n${underline}${bold}${blue}Note:${reset} ${blue}%s${reset}\n" "$@"
+}
+
+
+#--------------------------------------------------------------------------
+# 
+# Main script logic
+#
+#--------------------------------------------------------------------------
+
+
+#--------------------------------------------------------------------------
+# Check that the ../framework is present.
+h2 "Making sure the openapi yaml file is available..."
+if [[ ! -e "../framework" ]]; then
+    error "../framework is not present. Clone the framework repository."
+    info "The openapi.yaml file from the framework repository is needed to generate a go client for the rest API"
+    exit 1
+fi
+
+if [[ ! -e "../framework/openapi.yaml" ]]; then 
+    error "File ../framework/openapi.yaml is not found."
+    info "The openapi.yaml file from the framework repository is needed to generate a go client for the rest API"
+    exit 1
+fi
+success "OK"
+
+#--------------------------------------------------------------------------
+# Create a temporary folder which is never checked in.
+h2 "Making sure the tools folder is present."
+mkdir -p tools
+rc=$? ; if [[ "${rc}" != "0" ]]; then error "Failed to ensure the tools folder is present. rc=${rc}" ; exit 1 ; fi
+success "OK"
+
+#--------------------------------------------------------------------------
+# Download the open api generator tool if we've not got it already.
+export OPENAPI_GENERATOR_CLI_VERSION="6.2.0"
+export OPENAPI_GENERATOR_CLI_JAR=${BASEDIR}/tools/openapi-generator-cli-${OPENAPI_GENERATOR_CLI_VERSION}.jar
+if [[ ! -e ${OPENAPI_GENERATOR_CLI_JAR} ]]; then
+    info "The openapi generator tool is not available, so download it."
+    export OPENAPI_GENERATOR_CLI_SITE="https://repo1.maven.org/maven2/org/openapitools/openapi-generator-cli"
+    wget ${OPENAPI_GENERATOR_CLI_SITE}/${OPENAPI_GENERATOR_CLI_VERSION}/openapi-generator-cli-${OPENAPI_GENERATOR_CLI_VERSION}.jar \
```

Comment:
May be picky but is there any way to do this part without having wget installed? As I don't, so installing it before running this.

Reply:
> The output looks great. I do agree with the desire to remove the framework clone component from the build process. Should be a retrievable definition rather than a required clone. Especially since the yaml isn't used in the framework build currently (to the best of my knowledge).

runs, cps, webui and ras all seem to take the openapi.yaml file and generate client code from it ? They all contain this kind of thing:
```
openApiGenerate {
    generatorName = "typescript-rxjs"
    inputSpec = "$projectDir/openapi.yaml".toString()
    outputDir = "$buildDir/openapi/".toString()
    additionalProperties = [
        npmName: "galasa-runs-api-ts-rxjs",
        npmRepository: System.getenv('NPM_REPO'),
        npmVersion: System.getenv('NPM_VERSION'),
        snapshot: System.getenv('NPM_SNAPSHOT'),
        supportsES6: "false",
        modelPropertyNaming: "original"
    ]
}
```